File: hackerrank/easy/0042-equality-in-a-array/python/solution.py
## solution.py

from collections import Counter

def equalize_array(arr):
    freq = Counter(arr)
    max_freq = max(freq.values())
    return len(arr) - max_freq


# --- demo runs ---
if __name__ == "__main__":
    print(equalize_array([3, 3, 2, 1, 3]))  # 2
    print(equalize_array([1, 2, 2, 3]))     # 2

File: hackerrank/easy/0042-equality-in-a-array/js/solution.mjs
// solution.mjs

export function equalizeArray(arr) {
  const freq = {};
  for (const num of arr) {
    freq[num] = (freq[num] || 0) + 1;
  }
  const maxFreq = Math.max(...Object.values(freq));
  return arr.length - maxFreq;
}

// --- demo runs ---
console.log(equalizeArray([3, 3, 2, 1, 3])); // 2
console.log(equalizeArray([1, 2, 2, 3]));    // 2